### ISSUE TITLE
Expose errors 

### ISSUE THREAD

**Issue by abidlabs** [gradio-app/gradio-discord-bot#4]
Instead of not replying, it would be better if errors were exposed

**Comment by SaboniAmine** [comment]
Which level of error would you like to expose ?
For instance, trying to query the `stabilityai/stable-diffusion` space, I get a gradio internal error : 
```
    serialized_input = block.serialize(inputs[i])
AttributeError: 'Dataset' object has no attribute 'serialize'
```
But when querying the mirrored `abidlabs/images` space I might wait a long time to get no answer (understandable, with the current queue), or explicitely get an ` asyncio.exceptions.CancelledError` much sooner.
The second might be caught and sent back to the channel as a timeout, but I don't know if users should be exposed to the first one as is.

**Comment by abidlabs** [comment]
Good point @SaboniAmine! How did you see the gradio internal error when you tried to query `stabilityai/stable-diffusion` space?

**Comment by abidlabs** [comment]
In terms of the second error, I think we should also expose the queue ETA when people first make a request to a particular Space. However, this should probably be a separate issue. Would you like to go ahead and create it?
